### PR TITLE
global sort: raise maximum overlapping threshold

### LINKED ISSUE #50293: 100T Google Cloud Storage global sort failed because the file number is larger than expected
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: close #50293

Problem Summary:

### What changed and how does it work?

- as tile
- batch the calculating of startOffsets and endOffsets into one seekPropsOffsets call
- enlarge buffer for stat reader and enable prefetch for stat reader

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

## manual tests
- manually run 1T test with 3 tidb 3 tikv. Before this PR, master branch 1h40min. With this PR 1h45min.

- `go test ./br/pkg/lightning/backend/external  -test.run "Test3KFilesRangeSplitter" -testing-storage-uri "gs://xxx" -test.v --tags=intest`, cost 17 minutes to finish the RangeSplitter. Maximum memory usage 22G.

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/lightning/backend/external/engine.go [hunk 1]
```diff
@@ -293,9 +293,9 @@ func (e *Engine) LoadIngestData(
 	regionRanges []common.Range,
 	outCh chan<- common.DataAndRange,
 ) error {
-	// currently we assume the region size is 96MB and will download 96MB*40 = 3.8GB
+	// currently we assume the region size is 96MB and will download 96MB*20 = 1.9GB
 	// data at once
-	regionBatchSize := 40
+	regionBatchSize := 20
```
Comment (lance6716) [inline br/pkg/lightning/backend/external/engine.go:298]: I have tested that only with this change, performance won't drop and OOM can be avoided with "thread=8" SQL parameter.

Comment (lance6716) [reply br/pkg/lightning/backend/external/engine.go:298]: In this week when I run the 1T test with thread=8 and thread=16 arguments, thread=8 will OOM more frequently.

Considering that before this PR, merge step will loop for every 1000 files merging them into 8 or 16 files. When thread=8, one output file is merged from 125 files, when thread=16, one output file is merged from 63 files. OOM may cause by the larger ourput file size. And lower this batch size will reduce the chance of OOM.

Now we are going to let the merge step loop for 4000 files, with thread=16, one output file is merged from 250 files. So it's more likely that OOM will occur. I'm running the 1T test to check if the batch size is lower enough.

**Review by D3Hunter — COMMENTED** [review]
rest lgtm

**Review by D3Hunter — APPROVED** [review]

**Review by Benjamin2037 — APPROVED** [review]
LGTM

**Review by Leavrth — APPROVED** [review]

**Pull request merged**